### PR TITLE
Add Remaining Gossip Changes

### PR DESCRIPTION
**Pull request by nisdas**
**What type of PR is this?**

Altair Work

**What does this PR do? Why is it needed?**

- [x] Allows our pubsub decoder to correctly process and handle different block types.
- [x] Adds changes to allow for multiple block types in our beacon block subscriber and validator.
- [x] Removes some changes for our attestation validator so as to now contain all logic in the pubsub decoder.
- [x] Fix references to tests across the repo.

**Which issues(s) does this PR fix?**

Fixes #

**Other notes for review**

### CODE REVIEW

File: beacon-chain/sync/decode_pubsub.go [hunk 1]
```diff
@@ -8,6 +8,7 @@ import (
 	pubsub "github.com/libp2p/go-libp2p-pubsub"
 	"github.com/pkg/errors"
 	"github.com/prysmaticlabs/prysm/beacon-chain/p2p"
+	eth "github.com/prysmaticlabs/prysm/proto/prysm/v1alpha1"
```
Comment (rauljordan) [inline beacon-chain/sync/decode_pubsub.go:11]: Duplicate import. We tend to use `eth` to refer to v1 stuff

**Review by rauljordan — APPROVED** [review]

File: beacon-chain/sync/decode_pubsub.go [hunk 2]
```diff
@@ -43,6 +47,13 @@ func (s *Service) decodePubsubMessage(msg *pubsub.Message) (ssz.Unmarshaler, err
 	if !ok {
 		return nil, errors.Errorf("message of %T does not support marshaller interface", base)
 	}
+	// Handle different message types across forks.
+	if topic == p2p.BlockSubnetTopicFormat {
```
Comment (prestonvanloon) [inline beacon-chain/sync/decode_pubsub.go:51]: How does this equality work? BlockSubnetTopicFormat is `/eth/%x/beacon_block`, but wouldn't topic include the digest?

Comment (prestonvanloon) [reply beacon-chain/sync/decode_pubsub.go:51]: Ah, nevermind, we replace the topic digest with the formatting.

File: beacon-chain/sync/subscriber_sync_committee_message.go [hunk 3]
```diff
@@ -9,6 +9,7 @@ import (
 	"google.golang.org/protobuf/proto"
 )
 
+// skipcq: SCC-U1000
```
Comment (prestonvanloon) [inline beacon-chain/sync/subscriber_sync_committee_message.go:12]: What is this?

File: beacon-chain/sync/validate_beacon_blocks.go [hunk 4]
```diff
@@ -145,7 +142,7 @@ func (s *Service) validateBeaconBlockPubSub(ctx context.Context, pid peer.ID, ms
 	}
 
 	// Handle block when the parent is unknown.
-	if !s.cfg.DB.HasBlock(ctx, bytesutil.ToBytes32(blk.Block().ParentRoot())) {
+	if !s.cfg.DB.HasBlock(ctx, bytesutil.ToBytes32(blk.Block().ParentRoot())) && !s.cfg.Chain.HasInitSyncBlock(bytesutil.ToBytes32(blk.Block().ParentRoot())) {
```
Comment (terencechain) [inline beacon-chain/sync/validate_beacon_blocks.go:145]: This seems not trivial

What's the rationale for adding `!s.cfg.Chain.HasInitSyncBlock(bytesutil.ToBytes32(blk.Block().ParentRoot()))` ?

If it's not a hard fork specific change, should we have a discussion around it? perhaps another PR?

File: beacon-chain/sync/subscriber_sync_committee_message.go (hunk 3 above)
Comment (prestonvanloon) [reply beacon-chain/sync/subscriber_sync_committee_message.go:12]: nvm, its a deepsource thing

**Review by prestonvanloon — DISMISSED** [review]

File: beacon-chain/sync/subscriber_sync_committee_message.go (hunk 3 above)
Comment (rauljordan) [reply beacon-chain/sync/subscriber_sync_committee_message.go:12]: Unused code to satisfy deep source for now, probably can be removed

**Review by terencechain — CHANGES_REQUESTED** [review]
The additional of `s.cfg.Chain.HasInitSyncBlock(bytesutil.ToBytes32(blk.Block().ParentRoot()))` needs to be further reviewed

File: beacon-chain/sync/validate_beacon_blocks.go (hunk 4 above)
Comment (nisdas) [reply beacon-chain/sync/validate_beacon_blocks.go:145]: Lets remove it from this PR then, I can add it in another one. Without this change a node can get permanently stuck when finally processing blocks after syncing an extended chunk of non finalized blocks.

**Review by terencechain — APPROVED** [review]

**Pull request merged**